### ISSUE TITLE
bug: When using custom auth provider in self-hosted version, I always get "No email found in user object"

### ISSUE THREAD

**Issue by mariusp12** [langfuse/langfuse#4310]
### Describe the bug

Hello,

When trying to deploy langfuse on a Kubernetes cluster with the following environment variables (related to auth):

        LANGFUSE_INIT_USER_EMAIL=John.Doe@mycompany.com
        LANGFUSE_INIT_ORG_ID=mycompany
        LANGFUSE_INIT_ORG_NAME=D1
        LANGFUSE_INIT_PROJECT_ID=myProject
        LANGFUSE_INIT_PROJECT_NAME=My Project
        LANGFUSE_INIT_PROJECT_PUBLIC_KEY: pk-lf-xxxxx
        AUTH_DISABLE_USERNAME_PASSWORD=true
        AUTH_CUSTOM_ISSUER=http://mycompany.com/oidc/issuer"
        AUTH_CUSTOM_NAME="My Company OIDC Issuer"
        AUTH_CUSTOM_CLIENT_ID=1234-5678-ddd
        AUTH_CUSTOM_SCOPE=openid email profile bmwids
        AUTH_CUSTOM_CLIENT_SECRET=client-id-secret
        
I get the following behavior: 
- I am redirected to the login page of the custom provider
- afterwards, the provider redirects to the correct URL (/api/auth/callback/custom)
- in the UI I see the following error message: "No email found in user object"

I use langfuse version 2.90.0

We make the calls to the OIDC provider via an Envoy. In the Envoy, however, there is no call to the userinfo endpoint, just to the .well-known/openid-configuration, which is retrieved correctly.

Is there a general problem with the custom OIDC providers or am I doing something wrong? 

Thank you!

### To reproduce

See description above

### SDK and container versions

2.90.0

### Are you interested to contribute a fix for this bug?

No

**Comment by mariusp12** [comment]
@dosu The email is only provided by the userinfo endpoint and not in the token. So the userinfo needs to be called. Is there any way to enforce this?

**Comment by mariusp12** [comment]
@dosu how can I modify the authorization flow? This is implemented by langfuse.

**Comment by marcklingen** [comment]
thanks for creating this issue! (and sorry for my delay to take this up)<br><br>We are open to contributions to the authentication logic, see troubleshooting hints here: [https://langfuse.com/docs/deployment/self-host#troubleshooting](https://langfuse.com/docs/deployment/self-host#troubleshooting)

Under the hood, this is the nextauth oidc connector, if you want to add more environment flags to configure it, please open a PR.

**Comment by marcklingen** [comment]
Does the user object include an email here?

**Comment by mariusp12** [comment]
@marcklingen  Yes, the user object includes the e-mail. The problem is that the userinfo endpoint does not get called.

After looking into the langfuse code and the nextauth code, the parameter needs to be set to false in order for the userinfo endpoint to be called (default in langfuse is always true: https://github.com/langfuse/langfuse/blob/23a4bbd2cbd38c0494c4d5ff0f727256a80173f2/packages/shared/src/server/auth/customSsoProvider.ts#L20). See documentation from nextauth here: https://next-auth.js.org/configuration/providers/oauth#options

So basically, langfuse would need to make the idtoken parameter configurable.

Are you still supporting langfuse v2? We haven't switched to v3 yet, but if it is possible to bring this change to v2, I could try to open a PR for it.

**Comment by marcklingen** [comment]
Thanks for sharing.

Yes, please open a PR from the `main` branch (v3). We can cherry-pick this to the v2 branch once this is merged if this is relevant for you.

**Comment by marcklingen** [comment]
Let me know if you have any questions while working on this, happy to help

**Comment by mariusp12** [comment]
@marcklingen  I am currently trying to create a branch. However, I don't have any rights to do that. Are there any rights which you need to give me, or am I doing something wrong?

**Comment by marcklingen** [comment]
You need to fork the repo and contribute a PR
https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project

**Comment by mariusp12** [comment]
@marcklingen I opened the following PR: https://github.com/langfuse/langfuse/pull/5193. Can you please have a look?

**Comment by marcklingen** [comment]
Thank you, I've just reviewed it and will merge this now

**Comment by mariusp12** [comment]
@marcklingen Thank you! Can you please also cherry-pick it to v2? Thank you!

**Comment by marcklingen** [comment]
> [@marcklingen](https://github.com/marcklingen) Thank you! Can you please also cherry-pick it to v2? Thank you!

I made a note to do this later today once this is on v3, I'd strongly encourage you to upgrade to v3 though to benefit from new features and bug fixes

**Comment by mariusp12** [comment]
> > [@marcklingen](https://github.com/marcklingen) Thank you! Can you please also cherry-pick it to v2? Thank you!
> 
> I made a note to do this later today once this is on v3, I'd strongly encourage you to upgrade to v3 though to benefit from new features and bug fixes

we have it on our to do list, but since it involves a lot of big changes, we will need to analyze it more thoroughly. Thank you for your support on this matter!

**Comment by marcklingen** [comment]
Also released in https://github.com/langfuse/langfuse/releases/tag/v2.93.9, docker image will be available in roughly 90 minutes
